src/vm_stack.zig

```diff
@@ -3469,11 +3475,11 @@ const InstructionFuncs = struct {
         const dest_offset_u32 = @as(u32, @intCast(dest_offset));
         const length_u32 = @as(u32, @intCast(length));
 
-        var source = buffer[source_offset_u32 .. source_offset_u32 + length_u32];
-        var destination = buffer[dest_offset_u32 .. dest_offset_u32 + length_u32];
+        const source = buffer[source_offset_u32 .. source_offset_u32 + length_u32];
+        const destination = buffer[dest_offset_u32 .. dest_offset_u32 + length_u32];
 
         if (@intFromPtr(destination.ptr) < @intFromPtr(source.ptr)) {
-            std.mem.copy(u8, destination, source);
+            std.mem.copyForwards(u8, destination, source);
```

Comment:
`@memcpy`?

Reply:
Actually scratch that, I think `memory.copy` can deal with overlapping memory regions, so `copyForwards` is the correct one here.